### ISSUE TITLE
Incompatible with @typescript-eslint

### ISSUE THREAD

**Issue by darthtrevino** [lukastaegert/eslint-plugin-tree-shaking#68]
When using the @typescript-eslint plugin in tandem with this package, there are `UnknownNodeType` errors

```sh
Unknown node type TSInterfaceDeclaration.
If you are using the latest version of this plugin, please consider filing an issue noting this message, the offending statement, your ESLint version, and any active ESLint presets and plugins  tree-shaking/no-side-effects-in-initialization
```

The issue appears to be [here](https://github.com/lukastaegert/eslint-plugin-tree-shaking/blob/2f1e8fa1d05e0ec09f6febfcc5ef6b137f4b2873/lib/rules/no-side-effects-in-initialization.js#L906) where you are checking the AST for known node types.

It looks like you should either add typescript specific AST nodes to the known node types, or maybe check for a `TS` prefix, or skip this check altogether and ignore unknown node types.

**Comment by JustFly1984** [comment]
+1 for typescript support

**Comment by lukastaegert** [comment]
Sorry for the long wait. This is addressed now by #70
